### PR TITLE
Create NotificationStore and test it

### PR DESCRIPTION
**Pull request by kennyt**
First step towards being able to notify the user something urgent (for example, having a badge with a number next to a tab)

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @philipnrmn

File: src/js/stores/NotificationStore.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        dcosSecretStore: 1,
+      },
+      'security-tab': {
+        dcosSecretStore: 1
+      },
+      'stores-tab': {
+        dcosSecretStore: 1
+      }
+    }
+  */
```
Comment (rcorral) [inline src/js/stores/NotificationStore.js]: Can we remove this comment?

File: src/js/stores/NotificationStore.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
+import {Store} from 'mesosphere-shared-reactjs';
+
+import {NOTIFICATION_CHANGE} from '../constants/EventTypes';
+
```
Comment (rcorral) [inline src/js/stores/NotificationStore.js]: Remove this line.

File: src/js/stores/__tests__/NotificationStore-test.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    it('emits the correct event on visiblityChange', function () {
+      expect(this.mockFn).toBeCalled();
+    });
+
+    it('sets the correct number', function () {
+      NotificationStore.addNotification('foo', 'bar', 1);
+      expect(NotificationStore.getNotificationCount('foo', 'bar')).toEqual(1);
+      NotificationStore.removeNotification('foo', 'bar');
+      expect(NotificationStore.getNotificationCount('foo', 'bar')).toEqual(0);
+    });
```
Comment (rcorral) [inline src/js/stores/__tests__/NotificationStore-test.js]: Can you add a test where you just call `#removeNotification`

File: src/js/stores/__tests__/NotificationStore-test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+var NotificationStore = require('../NotificationStore');
+
+describe('NotificationStore', function () {
+  beforeEach(function () {
+    this.mockFn = jest.genMockFunction();
+    NotificationStore.addChangeListener(NOTIFICATION_CHANGE, this.mockFn);
+    NotificationStore.addNotification('foo', 'bar', 1);
+  });
+
+  describe('#addNotification', function () {
```
Comment (rcorral) [inline src/js/stores/__tests__/NotificationStore-test.js]: You didn't write any explicit tests for `getNotificationCount`

File: src/js/stores/NotificationStore.js (hunk 1 above)
Comment (kennyt) [inline src/js/stores/NotificationStore.js]: done

File: src/js/stores/NotificationStore.js (hunk 2 above)
Comment (kennyt) [inline src/js/stores/NotificationStore.js]: done

File: src/js/stores/__tests__/NotificationStore-test.js (hunk 3 above)
Comment (kennyt) [inline src/js/stores/__tests__/NotificationStore-test.js]: done

File: src/js/stores/__tests__/NotificationStore-test.js (hunk 4 above)
Comment (kennyt) [inline src/js/stores/__tests__/NotificationStore-test.js]: wrote some

**Pull request merged**